core/src/test/java/org/elasticsearch/common/PriorityTests.java

```diff
@@ -82,11 +81,11 @@ public void testCompareTo() {
         for (Priority p : Priority.values()) {
             assertEquals(0, p.compareTo(p));
         }
-        List<Priority> shuffeledAndSorted = new ArrayList<>(Priority.values());
+        List<Priority> shuffeledAndSorted = Arrays.asList(Priority.values());
         Collections.shuffle(shuffeledAndSorted, random());
         Collections.sort(shuffeledAndSorted);
         for (List<Priority> priorities : Arrays.asList(shuffeledAndSorted,
-            Priority.values())) { // #values() guarantees order!
+            Arrays.asList(Priority.values()))) { // #values() guarantees order!
```

Comment:
We should never rely on the ordinal anywhere, it's just a maintenance burden. The tests here do ensure that the values are in the right order though: `IMMEDIATE < URGENT < HIGH < NORMAL < LOW < LANGUID`. I think that's all we need?